### PR TITLE
Compat: Social Links: Remove legacy renderers from packages

### PR DESCRIPTION
**Pull request by mcsf**
## Description

As of #19887, Social Link is implemented as a single block type with multiple variations, whereas previously each supported service (e.g. WordPress, Twitter) was implemented as its own cloned block type:

Old format: `<!-- wp:social-link-wordpress {"url":"foo"} /-->`
New format: `<!-- wp:social-link {"service":"wordpress","url":"foo"} /-->`

However, Social Link is defined as a dynamic block. Thus, for backwards compatibility, #19887 preserved the server-side registration of all the "cloned" block types. However, this requirement only applies to the Gutenberg plugin, since Social Links haven't landed in core yet. Thus, this PR moves the server-side registration of cloned block types out of packages — specifically, `block-library` and into `lib/blocks.php`.

This will ensure that, the next time packages are published and pulled into WordPress core, no added work is needed to make sure that core installations do not register unneeded legacy block types.

## How has this been tested?
- Make sure that both new-format and old-format Social Link blocks are correctly rendered on the front-end.
- If the process were more ergonomic, it would be ideal to test that publishing packages and building core from them pulls in `core/social-link` but no `core/social-link-xyz` clones.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: lib/blocks.php [hunk 1]
```diff
@@ -85,6 +63,87 @@ function gutenberg_reregister_core_block_types() {
...
+					),
+					'service' => array(
+						'type'    => 'string',
+						'default' => $site,
+					),
+					'label'   => array(
+						'type' => 'string',
+					),
+				),
+				'render_callback' => 'gutenberg_render_core_social_link',
```
Comment (mcsf) [inline lib/blocks.php]: Will need rebasing after #20085.

File: lib/blocks.php [hunk 2]
```diff
@@ -85,6 +63,87 @@ function gutenberg_reregister_core_block_types() {
...
+ * plugin who have used Social Links prior to their conversion to block
+ * variations.
+ *
+ * This shim is INTENTIONALLY left out of core, as Social Links haven't yet
+ * landed there.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/19887
+ */
+function gutenberg_register_legacy_social_link_blocks() {
+	$sites = array(
```
Comment (gziolo) [inline lib/blocks.php]: It probably could be:
```suggestion
	$services = array(
```

but I see why it is called `$sites`. It's your call what to land.

**Review by gziolo — COMMENTED** [review]
Code-wise this is exactly what I expected to see 👍 A nice improvement that we can fully test next week after it is integrated with WordPress core.

File: lib/blocks.php (hunk 2 above)
Comment (mcsf) [reply lib/blocks.php]: That works — done.

File: lib/blocks.php [hunk 3]
```diff
@@ -85,6 +63,87 @@ function gutenberg_reregister_core_block_types() {
 }
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
 
+/**
+ * Complements the implementation of block type `core/social-icon`, whether it
```
Comment (gziolo) [inline lib/blocks.php]: I like where you are going with this, but I think we decided to leave the old name: 
```suggestion
 * Complements the implementation of block type `core/social-link`, whether it
```

😃

**Review by gziolo — APPROVED** [review]
I can confirm it works as expected. When I remove this code, I can't see Social Icons on the frontend with the legacy version saved in the post content. With this patch, everything works as before.

**Comment by mcsf** [comment]
Hah, thanks for wrapping this one up for me!

**Pull request merged**